testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2625,23 +2624,25 @@ public void testRenameColumn()
     }
 
     @Test
-    public void testSetColumnType()
-    {
-        if (!hasBehavior(SUPPORTS_SET_COLUMN_TYPE)) {
-            assertQueryFails("ALTER TABLE nation ALTER COLUMN nationkey SET DATA TYPE bigint", "This connector does not support setting column types");
-            return;
-        }
-
+    public void testSetColumnType() {
         skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
 
-        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_set_column_type_", "AS SELECT CAST(123 AS integer) AS col")) {
-            assertUpdate("ALTER TABLE " + table.getName() + " ALTER COLUMN col SET DATA TYPE bigint");
-
-            assertEquals(getColumnType(table.getName(), "col"), "bigint");
-            assertThat(query("SELECT * FROM " + table.getName()))
-                    .skippingTypesCheck()
-                    .matches("VALUES bigint '123'");
-        }
+        String tableName = "test_set_column_type";
+        assertUpdate("CREATE TABLE " + tableName + " (a bigint, b double NOT NULL, c varchar(50)) WITH (order_by=ARRAY['b'], engine = 'MergeTree')");
+
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN a SET DATA TYPE varchar(50)");
```

Comment:
Column `a` doesn't have any definition. The new doesn't ensure if `SET DATA TYPE` preserved `a` column definition correctly or not because it created the table with table level property. I would recommend creating a table on Clickhouse so that we can specify column definitions (default_expr, codec, TTL).